ios/DomainLayer/Utilities/Sources/Utilities/Extensions/Data+Extensions.swift

```diff
@@ -10,8 +10,14 @@ public extension Data {
         if let keyPath = keyPath {
             let toplevel = try JSONSerialization.jsonObject(with: self)
             if let nestedJson = (toplevel as AnyObject).value(forKeyPath: keyPath) {
-                let nestedJsonData = try JSONSerialization.data(withJSONObject: nestedJson)
-                return try JSONDecoder().decode(D.self, from: nestedJsonData)
+                if JSONSerialization.isValidJSONObject(nestedJson) {
+                    let nestedJsonData = try JSONSerialization.data(withJSONObject: nestedJson)
+                    return try JSONDecoder().decode(D.self, from: nestedJsonData)
+                } else {
+                    let wrappedJsonObject = ["value": nestedJson]
```

Comment:
Shouldn't `"value"` be a parameter of this map function in case it will be something else in other responses? You can provide `"value"` as default value for it.

Reply:
This was taken from the MoyaProvider. Else statement occurs only when there are simple data type, so that's the reason to wrap it into "value" dictionary

Reply:
Got it, thx